src/ConnectionPool.jl

```diff
@@ -58,15 +55,15 @@ Fields:
 - `readable`, whether the Connection object is readable
 - `writable`, whether the Connection object is writable
 """
-mutable struct Connection <: IO
+mutable struct Connection{IO_t <: IO} <: IO
```

Comment:
this now make `Connection` a non-concrete type, which could be problematic if we ever have a collection of `Connection`s -- i see this PR address one such case by splitting `POOL` into two different `TCP_POOL` and `SSL_POOL` -- but i wonder whether or not `Connection` is public API and if we need to check this doesn't affect users? I _think_ it's not public API and we should be fine, but i know packages adding type params has caused me many headaches in the past

Reply:
Good points, yeah. Looking at the docstring, `io` is meant to be one of the two types, but it is definitely a good idea to check how breaking this is.

Reply:
I did a quick scan through AWS.jl, Azure.jl, RAICode, Pluto.jl, Genie.jl and Github.jl and it seems that no one is really touching `Connection`s. It is also not exported by HTTP.jl. I'm hopeful that most people won't notice this change, and those who do more involved work regarding connections might appreciate fewer allocs.

Reply:
great- thanks for checking!